ui/smite_priest/sim.ts

```diff
@@ -0,0 +1,241 @@
+import { RaidTarget } from '/tbc/core/proto/common.js';
+import { Spec } from '/tbc/core/proto/common.js';
+import { Stat } from '/tbc/core/proto/common.js';
+import { TristateEffect } from '/tbc/core/proto/common.js'
+import { Stats } from '/tbc/core/proto_utils/stats.js';
+import { Player } from '/tbc/core/player.js';
+import { Sim } from '/tbc/core/sim.js';
+import { IndividualSimUI } from '/tbc/core/individual_sim_ui.js';
+
+import { Alchohol} from '/tbc/core/proto/common.js';
+import { BattleElixir } from '/tbc/core/proto/common.js';
+import { Flask } from '/tbc/core/proto/common.js';
+import { Food } from '/tbc/core/proto/common.js';
+import { GuardianElixir } from '/tbc/core/proto/common.js';
+import { Conjured } from '/tbc/core/proto/common.js';
+import { Drums } from '/tbc/core/proto/common.js';
+import { PetFood } from '/tbc/core/proto/common.js';
+import { Potions } from '/tbc/core/proto/common.js';
+import { WeaponImbue } from '/tbc/core/proto/common.js';
+
+import { SmitePriest, SmitePriest_Rotation as Rotation, SmitePriest_Options as Options, SmitePriest_Rotation, SmitePriest_Rotation_RotationType } from '/tbc/core/proto/priest.js';
+
+import * as IconInputs from '/tbc/core/components/icon_inputs.js';
+import * as OtherInputs from '/tbc/core/components/other_inputs.js';
+import * as Mechanics from '/tbc/core/constants/mechanics.js';
+import * as Tooltips from '/tbc/core/constants/tooltips.js';
+
+import * as SmitePriestInputs from './inputs.js';
+import * as Presets from './presets.js';
+
+export class SmitePriestSimUI extends IndividualSimUI<Spec.SpecSmitePriest> {
+  constructor(parentElem: HTMLElement, player: Player<Spec.SpecSmitePriest>) {
+		super(parentElem, player, {
+			cssClass: 'smite-priest-sim-ui',
+			// List any known bugs / issues here and they'll be shown on the site.
+			knownIssues: [
+			],
+
+			// All stats for which EP should be calculated.
+			epStats: [
+				Stat.StatIntellect,
+				Stat.StatSpirit,
+				Stat.StatSpellPower,
+				Stat.StatSpellHit,
+				Stat.StatSpellCrit,
+				Stat.StatSpellHaste,
+				Stat.StatMP5,
+			],
+			// Reference stat against which to calculate EP. I think all classes use either spell power or attack power.
+			epReferenceStat: Stat.StatSpellPower,
+			// Which stats to display in the Character Stats section, at the bottom of the left-hand sidebar.
+			displayStats: [
+				Stat.StatStamina,
+				Stat.StatIntellect,
+				Stat.StatSpirit,
+				Stat.StatSpellPower,
+				Stat.StatShadowSpellPower,
+				Stat.StatHolySpellPower,
+				Stat.StatSpellHit,
+				Stat.StatSpellCrit,
+				Stat.StatSpellHaste,
+				Stat.StatMP5,
+			],
+			modifyDisplayStats: (player: Player<Spec.SpecSmitePriest>, stats: Stats) => {
+				return stats.withStat(Stat.StatSpellHit,
+						stats.getStat(Stat.StatSpellHit)
+						+ player.getTalents().shadowFocus * 2 * Mechanics.SPELL_HIT_RATING_PER_HIT_CHANCE
+						+ player.getTalents().focusedPower * 2 * Mechanics.SPELL_HIT_RATING_PER_HIT_CHANCE);
+			},
```

Comment:
I think you'll want to remove this function, so that shadowFocus and focusedPower don't look like regular hit in the character stats panel.

Reply:
I only appended focusedPower to what was existing for shadow. I don't disagree - but can you modify specific spell's hit based on the same formula instead?

Reply:
This function is controlling only the displayed +hit stat, and right now we don't have a visual for stats on individual spells. In most cases its not really worth having such a visual because the values can be complicated and you can see the actual hit rate in Detailed Results anyway. This was added specifically for cases like shadow priest, where the +hit is technically only for specific spells but in reality it applies to the whole kit that gets used.